### ISSUE TITLE
`while_let_on_iterator` lint is triggered when iterator is used on two levels

### ISSUE THREAD

**Issue by GuillaumeGomez** [rust-lang/rust-clippy#13602]
### Summary

In this code:

```rust
let mut iter = params.iter();
while let Some((arg_name, default_value) = iter.next() {
    if default_value.is_some() {
        while let Some((new_arg_name, default_value)) = iter.next() {
            // more code
        }
    }
}
```

unless I missed something, you cannot iterate on `iter` using a `for` loop. If I try to apply the lint suggestion, I get:

```
error[E0499]: cannot borrow `iter` as mutable more than once at a time
   --> rinja_parser/src/node.rs:640:58
    |
637 |             for (arg_name, default_value) in iter.by_ref() {
    |                                              -------------
    |                                              |
    |                                              first mutable borrow occurs here
    |                                              first borrow later used here
...
640 |                     for (new_arg_name, default_value) in iter.by_ref() {
    |                                                          ^^^^ second mutable borrow occurs here
```

### Lint Name

while_let_on_iterator

### Version

rust version 1.82.0 (f6e511eec 2024-10-15)

**Comment by Jarcho** [comment]
Clippy only suggests changing the inner loop which can be a for loop.

**Comment by GuillaumeGomez** [comment]
If I only update the inner loop I get:

```
error[E0382]: borrow of moved value: `iter`
   --> rinja_parser/src/node.rs:638:57
    |
636 |             let mut iter = params.iter();
    |                 -------- move occurs because `iter` has type `std::slice::Iter<'_, (&str, Option<WithSpan<'_, Expr<'_>>>)>`, which does not implement the `Copy` trait
637 |             #[allow(clippy::while_let_on_iterator)]
638 |             while let Some((arg_name, default_value)) = iter.next() {
    |             --------------------------------------------^^^^-------
    |             |                                           |
    |             |                                           value borrowed here after move
    |             inside of this loop
...
641 |                     for (new_arg_name, default_value) in iter {
    |                                                          ---- `iter` moved due to this implicit call to `.into_iter()`, in previous iteration of loop
```

**Comment by Jarcho** [comment]
The suggestion is to use `for (new_arg_name, default_value) in iter.by_ref()`, not using the iterator by value.
